### DISCUSSION TITLE
Fabric: Command Not Found

### DISCUSSION THREAD

**Discussion by Pride-Solutions-Derby** [danielmiessler/Fabric#950] [Q&A]
Hi All, anyone able to help? 

When installing the latest version of fabric using Go, on completion i try to run fabric --setup and i get back "fabric: command not found". 

Any ideas?

**Comment by icemarr** [comment]
Getting same issue!!

**Reply by Romilunas** [reply]
It may be (as it happened to me) that you have multiple Python environments, and the fabric command is in a version installed in a different environment (likely via pipx) if you just installed with pip  for instance. Uninstall with pipx uninstall fabric or whichever is not your latest env, activate the right env and try again.

**Comment by fonteijne** [comment]
Which OS and shell are you using?

I've faced this same issue after migrating from python to the GO version. 
Using **fish on MacOS**, had it solved by updating the $PATH:
```$ set -Ux PATH $PATH $GOPATH/bin```
If it returns:
```set: successfully set universal 'PATH'; but a global by that name shadows it```
use
```$ set -g PATH $PATH $HOME/go/bin```

To make it persistent, add the last line to `~/.config/fish/config.fish`
restart te terminal and try `$ fabric`.

**Reply by mfpaim** [reply]
I was having the same issue here and based on the fonteijne (thanks a lot). My 2 cents is below and this solved the issue for my **Mac using zsh:**
```
ls $HOME/go/bin
``` 
```
export PATH=$HOME/go/bin:$PATH
``` 
Then test:
```
fabric -h
``` 
Make It Permanent in Zsh
```
echo 'export PATH=$HOME/go/bin:$PATH' >> ~/.zshrc
``` 

```
source ~/.zshrc
``` 
Verify:
```
which fabric
``` 
enjoy:
```
fabric -h
``` 
Voilá!

**Comment by bledburn** [comment]
> Hi All, anyone able to help?
> 
> When installing the latest version of fabric using Go, on completion i try to run fabric --setup and i get back "fabric: command not found".
> 
> Any ideas?

Without getting more specific details, I'd say the previous responses to you issue are probably accurate. The fabric cli tool, after being installed using Go, now resides on your system as an "executabel" or "binary." However, when building with Go, the tool is inserted into the "Go-bin" by default (normally at ~/go/bin or something like that). As mentioned by mfpaim, putting the following `export PATH=$HOME/go/bin:$PATH` into your .bashrc or .bash_aliases file in your user's home directory and restarting your terminal session, will likely resolve the issue. Otherwise moving the fabric tool to any other location that is inserted to your system's PATH will solve this issue also. If, for some reason, you do not want to mess with your PATH, you can always create an alias in your .bashrc or .bash_aliases (or if you are using zsh it would be your .zshrc file).

alias fabric='~/go/bin/fabric'  #assuming this is where Go is installed, and where the Go-bin is located, and fabric is within it

Understanding your PATH is important, explained here: https://www.youtube.com/watch?v=hnFJhubYS88

Make sure Go installed successfully and it installed Fabric correctly, explained here: https://www.youtube.com/watch?v=_QMYQtYUpys
